pkg/cmd/azurekeyvaulttemplatefuncs.go

```diff
@@ -0,0 +1,97 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/security/keyvault/azsecrets"
+)
+
+const azureVaultDomain = "vault.azure.net"
+
+type azureKeyVault struct {
+	Name   string `json:"name"  mapstructure:"name"  yaml:"name"`
+	Alias  string `json:"alias" mapstructure:"alias" yaml:"alias"`
+	client *azsecrets.Client
+	cache  map[string]string
+}
+
+func (v *azureKeyVault) URL() string {
+	return "https://" + v.Name + "." + azureVaultDomain + "/"
+}
+
+type azureKeyVaultConfig struct {
+	Vaults []*azureKeyVault `json:"vaults" mapstructure:"vaults" yaml:"vaults"`
+	cred   *azidentity.DefaultAzureCredential
+}
+
+func (a *azureKeyVaultConfig) GetVaultNameFromAlias(alias string) string {
+	for _, vault := range a.Vaults {
```

Comment:
Switching to `map[string]*azureKeyVault` means that we can do named lookup rather than having to loop over and append. It’s a small efficiency, but I think worthwhile.

Reply:
Removed this function as not required anymore.